projects/mci/src/mci_broadcast.c

```diff
@@ -80,6 +80,27 @@ static void prv_broadcast_dsp_temp(MotorControllerBroadcastStorage *storage) {
                                (uint32_t)measurements[RIGHT_MOTOR_CONTROLLER].dsp_temp_c);
 }
 
+static void prv_temperature_fan_toggle(MotorControllerBroadcastStorage *storage) {
```

Comment:
I think that it might be better if the fan activation check (overtemp conditions) is handled in the functions where the data itself is processed (ie. `prv_handle_dsp_temp_rx` and `prv_handle_motor_sink_temp_rx`). I think this helps us to better track the cause and effect of data we're reading, and activate the fan right away as we reach the temp condition. 

The deactivation logic makes sense to have in this function, called periodically, as you need to know that all are below the off threshold to make the change

That being said, this solution works fine, and if there are other variables, such as this makes testing easier, then this is fine. Just something to think about, don't worry about changing